src/replayq.erl

```diff
@@ -419,11 +434,13 @@ pop_mem(
     StopFun,
     StopFunAcc
 ) ->
-    case queue:out(InMem) of
-        {{value, ?MEM_ONLY_ITEM(Sz, _Item)}, _} when
+    case replayq_mem:out(MemQueueModule, InMem) of
+        {{value, ?MEM_ONLY_ITEM(Sz, _) = MI}, InMem1} when
             BytesMode == at_most andalso Sz > Bytes andalso Acc =/= []
         ->
-            {Q, ?NOTHING_TO_ACK, lists:reverse(Acc)};
+            %% put the item back to the queue in reverse order
+            InMem2 = replayq_mem:in_r(MemQueueModule, MI, InMem1),
```

Comment:
Since we disregard the item here, there's no need to re-insert it into the memory queue, just keep the old `InMem` reference (same in the disk fn below).

Reply:
planning to implement `replayq_mem_ets` which deletes the item in the `out` API.